pkg/devfile/generator/generators.go

```diff
@@ -184,7 +184,7 @@ func GetDeployment(devfileObj parser.DevfileObj, deployParams DeploymentParams)
 	deployment := &appsv1.Deployment{
 		TypeMeta:   deployParams.TypeMeta,
 		ObjectMeta: deployParams.ObjectMeta,
-		Spec:       *getDeploymentSpec(deploySpecParams),
+		Spec:       *getDeploymentSpec(deploySpecParams, deployParams),
```

Comment:
add it to be an additional field in `deploySpecParams` instead of changing the function input

Reply:
@yangcao77 thanks, I have made the change.

@yangcao77 @feloy What surprised me was I no longer needed the `if` check. I was expecting a nil pointer error, but it seems that was my lack of understanding of Go.